src/infrastructure/db/sqlite/util.rs

```diff
@@ -98,38 +98,54 @@ impl From<e::Tag> for Tag {
 
 impl From<User> for e::User {
     fn from(u: User) -> e::User {
+        use num_traits::FromPrimitive;
         let User {
             id,
             username,
             password,
             email,
             email_confirmed,
+            role,
         } = u;
         e::User {
             id,
             username,
             password,
             email,
             email_confirmed,
+            role: e::Role::from_i16(role).unwrap_or_else(|| {
+                warn!(
+                    "Could not cast role from i16 (value: {}). Use {:?} instead.",
+                    role,
+                    e::Role::Guest
```

Comment:
Add a constant _default user role_ for this purpose? Currently `Role::Guest` is implicitly used as the user's default role in many places. The constant would associate it explicitly with the entity.